MonsterTrainModdingAPI/Managers/CustomTriggerManager.cs

```diff
@@ -3,65 +3,179 @@
+            return RegisterName;
+        }
+        /// <summary>
+        /// Registers a Localization String
+        /// </summary>
+        /// <typeparam name="T">The Card Trigger Type used to register for</typeparam>
+        /// <param name="RegisterName">The string to register for localization</param>
+        /// <returns></returns>
+        public static string RegisterCardLocalizationString<T>(string RegisterName = "") where T : IMTCardTrigger
+        {
+            if (RegisterName == "") RegisterName = "Trigger_" + typeof(T).Name;
+            Dictionary<CardTriggerType, string> dict = (Dictionary<CardTriggerType, string>)typeof(CardTriggerTypeMethods).GetField("TriggerToLocalizationExpression", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Static).GetValue(null);
+            dict[GetCardTrigger(typeof(T))] = RegisterName;
+            CardTriggerToNameDict[GetCardTrigger(typeof(T))] = RegisterName;
             return RegisterName;
         }
         /// <summary>
         /// Returns a Trigger by Integer ID
         /// </summary>
         /// <param name="ID">Integer to cast to Trigger</param>
         /// <returns></returns>
-        public static CharacterTriggerData.Trigger GetTrigger(int ID)
+        public static CharacterTriggerData.Trigger GetCharacterTrigger(int ID)
         {
             return (CharacterTriggerData.Trigger)ID;
         }
         /// <summary>
+        /// Returns a Trigger by Integer ID
+        /// </summary>
+        /// <param name="ID">Integer to cast to CardTriggerType</param>
+        /// <returns></returns>
+        public static CardTriggerType GetCardTrigger(int ID)
+        {
+            return (CardTriggerType)ID;
+        }
+        /// <summary>
         /// Returns a Trigger by an IMT reference
         /// </summary>
         /// <param name="data">Data to get Integer ID</param>
         /// <returns></returns>
-        public static CharacterTriggerData.Trigger GetTrigger(IMTCharacterTrigger data)
+        public static CharacterTriggerData.Trigger GetCharacterTrigger(IMTCharacterTrigger data)
         {
-            return GetTrigger(data.ID);
+            return GetCharacterTrigger(data.ID);
+        }
+        /// <summary>
+        /// Returns a CardTriggerType by an IMT reference
+        /// </summary>
+        /// <param name="data"></param>
+        /// <returns>Data to get enum ID</returns>
+        public static CardTriggerType GetCardTrigger(IMTCardTrigger data)
+        {
+            return GetCardTrigger(data.ID);
         }
         /// <summary>
         /// Returns a Trigger by Type
         /// </summary>
-        /// <param name="type">Type to get the ID from</param>
+        /// <param name="type">Type to get the ID from, should inherit from IMTCharacterTrigger</param>
+        /// <returns></returns>
+        public static CharacterTriggerData.Trigger GetCharacterTrigger(Type type)
```

Comment:
There are a lot of overloads for this method. My opinion: having a ton of different overloads makes it confusing for developers, since you aren't sure which one you "should" be using. Do we really need all these? Yes, you "can" use any of them, but typically when several overloads are provided that accomplish the same functionality, only one ever actually gets used.

Again, my opinion. I'd be interested in discussing this as a group later.